diff --git a/vendor/magento/module-catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php b/vendor/magento/module-catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php
index b32c34678fe..ce097dd95d7 100644
--- a/vendor/magento/module-catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php
+++ b/vendor/magento/module-catalog/Model/ResourceModel/Product/Indexer/Eav/Source.php
@@ -18,7 +18,7 @@ use Magento\Framework\DB\Sql\UnionExpression;
  */
 class Source extends AbstractEav
 {
-    const TRANSIT_PREFIX = 'transit_';
+    public const TRANSIT_PREFIX = 'transit_';
 
     /**
      * Catalog resource helper
@@ -441,7 +441,7 @@ class Source extends AbstractEav
         $data = [];
         $query = $select->query();
         while ($row = $query->fetch()) {
-            $values = explode(',', $row['value']);
+            $values = isset($row['value']) ? explode(',', $row['value']) : [];
             foreach ($values as $valueId) {
                 if (isset($options[$row['attribute_id']][$valueId])) {
                     $data[] = [$row['entity_id'], $row['attribute_id'], $row['store_id'], $valueId, $row['source_id']];
