diff --git a/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/api.js b/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/api.js
index 232280b..fbf6ba3 100644
--- a/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/api.js
+++ b/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/api.js
@@ -124,7 +124,7 @@ define([
             };
 
             if (this.getEnvironment() !== "TEST") {
-                result['merchantId'] = this.getMerchantId();
+                result.merchantInfo = { merchantId: this.getMerchantId() };
             }
 
             return result;
diff --git a/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/implementations/core-checkout/method-renderer/googlepay.js b/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/implementations/core-checkout/method-renderer/googlepay.js
index f686297..59121a7 100644
--- a/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/implementations/core-checkout/method-renderer/googlepay.js
+++ b/vendor/paypal/module-braintree-core/view/frontend/web/js/googlepay/implementations/core-checkout/method-renderer/googlepay.js
@@ -91,7 +91,7 @@ define([
             };
 
            if (this.getEnvironment() !== "TEST") {
-               result['merchantId'] = this.getMerchantId();
+               result.merchantInfo = { merchantId: this.getMerchantId() };
            }
 
            return result;
