diff --git a/vendor/klevu/module-productsearch/Framework/Request/Cleaner.php b/vendor/klevu/module-productsearch/Framework/Request/Cleaner.php
index 22068d5f8..04e9b69ca 100644
--- a/vendor/klevu/module-productsearch/Framework/Request/Cleaner.php
+++ b/vendor/klevu/module-productsearch/Framework/Request/Cleaner.php
@@ -5,6 +5,7 @@
 namespace Klevu\Search\Framework\Request;
 
 use Magento\Framework\Search\Request\Aggregation\StatusInterface as AggregationStatus;
+use Magento\Framework\Search\Request\EmptyRequestDataException;
 use Magento\Framework\Session\SessionManagerInterface;
 use Magento\Framework\App\Config\MutableScopeConfigInterface as MutableScopeConfigInterface;
 use Magento\Framework\Registry as MagentoRegistry;
@@ -70,8 +71,13 @@ class Cleaner extends \Magento\Framework\Search\Request\Cleaner
      */
     public function clean(array $requestData)
     {
-        $requestData = parent::clean($requestData);
+        try {
+            $requestData = parent::clean($requestData);
+        } catch(EmptyRequestDataException $th){
+            // Ignore exception about empty query and filter because it handled further
+        }
         $requestData = $this->klevuQueryCleanup($requestData);
+
         return $requestData;
     }
 
@@ -111,6 +117,7 @@ class Cleaner extends \Magento\Framework\Search\Request\Cleaner
         $categoryFilter = isset($categoryFilter) ? $categoryFilter : '';
         // check if we are in search page
         if(!isset($requestData['queries']['quick_search_container'])) return $requestData;
+        if(!isset($requestData['queries']['search'])) return $requestData;
         //check if klevu is supposed to be on
         if ($this->klevuConfig->isLandingEnabled()!=1 || !$this->klevuConfig->isExtensionConfigured()) return $requestData;
         //save data in session so we do not request via api for filters
