diff --git a/vendor/magento/module-target-rule/Model/Rule/Condition/Product/Attributes/SqlBuilder.php b/vendor/magento/module-target-rule/Model/Rule/Condition/Product/Attributes/SqlBuilder.php
index 816c3b0d66c..0cad50e93dc 100644
--- a/vendor/magento/module-target-rule/Model/Rule/Condition/Product/Attributes/SqlBuilder.php
+++ b/vendor/magento/module-target-rule/Model/Rule/Condition/Product/Attributes/SqlBuilder.php
@@ -306,7 +306,12 @@ class SqlBuilder
             \Magento\Framework\DB\Select::SQL_UNION
         );
 
-        return 'e.' . $linkField . ' IN (' . $resultSelect . ')';
+        /**
+         * MySQL Subquery with IN statement performance optimizer
+         */
+        $selectWrapper = $this->indexResource->getConnection()->select()->from($resultSelect);
+
+        return 'e.' . $linkField . ' IN (' . $selectWrapper . ')';
     }
 
     /**
