diff --git a/vendor/magento/module-catalog-rule/Model/Indexer/IndexBuilder.php b/vendor/magento/module-catalog-rule/Model/Indexer/IndexBuilder.php
index 98b33e3d513..421a6e718b3 100644
--- a/vendor/magento/module-catalog-rule/Model/Indexer/IndexBuilder.php
+++ b/vendor/magento/module-catalog-rule/Model/Indexer/IndexBuilder.php
@@ -7,6 +7,7 @@
 namespace Magento\CatalogRule\Model\Indexer;
 
 use Magento\Catalog\Model\Product;
+use Magento\CatalogRule\Model\ResourceModel\Rule\Collection as RuleCollection;
 use Magento\CatalogRule\Model\ResourceModel\Rule\CollectionFactory as RuleCollectionFactory;
 use Magento\CatalogRule\Model\Rule;
 use Magento\Framework\App\ObjectManager;
@@ -15,6 +16,8 @@ use Magento\CatalogRule\Model\Indexer\IndexBuilder\ProductLoader;
 use Magento\CatalogRule\Model\Indexer\IndexerTableSwapperInterface as TableSwapper;
 
 /**
+ * Catalog rule index builder
+ *
  * @api
  * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
  * @SuppressWarnings(PHPMD.ExcessiveClassComplexity)
@@ -270,14 +273,14 @@ class IndexBuilder
      */
     protected function doReindexByIds($ids)
     {
-        $this->cleanByIds($ids);
+        $this->cleanProductIndex($ids);
 
         $products = $this->productLoader->getProducts($ids);
-        foreach ($this->getActiveRules() as $rule) {
-            foreach ($products as $product) {
-                $this->applyRule($rule, $product);
-            }
+        $activeRules = $this->getActiveRules();
+        foreach ($products as $product) {
+            $this->applyRules($activeRules, $product);
         }
+        $this->reindexRuleGroupWebsite->execute();
     }
 
     /**
@@ -322,6 +325,30 @@ class IndexBuilder
         );
     }
 
+    /**
+     * Clean product index
+     *
+     * @param array $productIds
+     * @return void
+     */
+    private function cleanProductIndex(array $productIds): void
+    {
+        $where = ['product_id IN (?)' => $productIds];
+        $this->connection->delete($this->getTable('catalogrule_product'), $where);
+    }
+
+    /**
+     * Clean product price index
+     *
+     * @param array $productIds
+     * @return void
+     */
+    private function cleanProductPriceIndex(array $productIds): void
+    {
+        $where = ['product_id IN (?)' => $productIds];
+        $this->connection->delete($this->getTable('catalogrule_product_price'), $where);
+    }
+
     /**
      * Clean by product ids
      *
@@ -330,51 +357,35 @@ class IndexBuilder
      */
     protected function cleanByIds($productIds)
     {
-        $query = $this->connection->deleteFromSelect(
-            $this->connection
-                ->select()
-                ->from($this->resource->getTableName('catalogrule_product'), 'product_id')
-                ->distinct()
-                ->where('product_id IN (?)', $productIds),
-            $this->resource->getTableName('catalogrule_product')
-        );
-        $this->connection->query($query);
-
-        $query = $this->connection->deleteFromSelect(
-            $this->connection->select()
-                ->from($this->resource->getTableName('catalogrule_product_price'), 'product_id')
-                ->distinct()
-                ->where('product_id IN (?)', $productIds),
-            $this->resource->getTableName('catalogrule_product_price')
-        );
-        $this->connection->query($query);
+        $this->cleanProductIndex($productIds);
+        $this->cleanProductPriceIndex($productIds);
     }
 
     /**
+     * Assign product to rule
+     *
      * @param Rule $rule
      * @param Product $product
-     * @return $this
-     * @throws \Exception
-     * @SuppressWarnings(PHPMD.NPathComplexity)
+     * @return void
      */
-    protected function applyRule(Rule $rule, $product)
+    private function assignProductToRule(Rule $rule, Product $product): void
     {
-        $ruleId = $rule->getId();
-        $productEntityId = $product->getId();
-        $websiteIds = array_intersect($product->getWebsiteIds(), $rule->getWebsiteIds());
-
         if (!$rule->validate($product)) {
-            return $this;
+            return;
         }
 
+        $ruleId = (int) $rule->getId();
+        $productEntityId = (int) $product->getId();
+        $ruleProductTable = $this->getTable('catalogrule_product');
         $this->connection->delete(
-            $this->resource->getTableName('catalogrule_product'),
+            $ruleProductTable,
             [
-                $this->connection->quoteInto('rule_id = ?', $ruleId),
-                $this->connection->quoteInto('product_id = ?', $productEntityId)
+                'rule_id = ?' => $ruleId,
+                'product_id = ?' => $productEntityId,
             ]
         );
 
+        $websiteIds = array_intersect($product->getWebsiteIds(), $rule->getWebsiteIds());
         $customerGroupIds = $rule->getCustomerGroupIds();
         $fromTime = strtotime($rule->getFromDate());
         $toTime = strtotime($rule->getToDate());
@@ -385,36 +396,44 @@ class IndexBuilder
         $actionStop = $rule->getStopRulesProcessing();
 
         $rows = [];
-        try {
-            foreach ($websiteIds as $websiteId) {
-                foreach ($customerGroupIds as $customerGroupId) {
-                    $rows[] = [
-                        'rule_id' => $ruleId,
-                        'from_time' => $fromTime,
-                        'to_time' => $toTime,
-                        'website_id' => $websiteId,
-                        'customer_group_id' => $customerGroupId,
-                        'product_id' => $productEntityId,
-                        'action_operator' => $actionOperator,
-                        'action_amount' => $actionAmount,
-                        'action_stop' => $actionStop,
-                        'sort_order' => $sortOrder,
-                    ];
-
-                    if (count($rows) == $this->batchCount) {
-                        $this->connection->insertMultiple($this->getTable('catalogrule_product'), $rows);
-                        $rows = [];
-                    }
+        foreach ($websiteIds as $websiteId) {
+            foreach ($customerGroupIds as $customerGroupId) {
+                $rows[] = [
+                    'rule_id' => $ruleId,
+                    'from_time' => $fromTime,
+                    'to_time' => $toTime,
+                    'website_id' => $websiteId,
+                    'customer_group_id' => $customerGroupId,
+                    'product_id' => $productEntityId,
+                    'action_operator' => $actionOperator,
+                    'action_amount' => $actionAmount,
+                    'action_stop' => $actionStop,
+                    'sort_order' => $sortOrder,
+                ];
+
+                if (count($rows) == $this->batchCount) {
+                    $this->connection->insertMultiple($ruleProductTable, $rows);
+                    $rows = [];
                 }
             }
-
-            if (!empty($rows)) {
-                $this->connection->insertMultiple($this->resource->getTableName('catalogrule_product'), $rows);
-            }
-        } catch (\Exception $e) {
-            throw $e;
         }
+        if ($rows) {
+            $this->connection->insertMultiple($ruleProductTable, $rows);
+        }
+    }
 
+    /**
+     * Apply rule
+     *
+     * @param Rule $rule
+     * @param Product $product
+     * @return $this
+     * @throws \Exception
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    protected function applyRule(Rule $rule, $product)
+    {
+        $this->assignProductToRule($rule, $product);
         $this->reindexRuleProductPrice->execute($this->batchCount, $product);
         $this->reindexRuleGroupWebsite->execute();
 
@@ -422,6 +441,25 @@ class IndexBuilder
     }
 
     /**
+     * Apply rules
+     *
+     * @param RuleCollection $ruleCollection
+     * @param Product $product
+     * @return void
+     */
+    private function applyRules(RuleCollection $ruleCollection, Product $product): void
+    {
+        foreach ($ruleCollection as $rule) {
+            $this->assignProductToRule($rule, $product);
+        }
+
+        $this->cleanProductPriceIndex([$product->getId()]);
+        $this->reindexRuleProductPrice->execute($this->batchCount, $product);
+    }
+
+    /**
+     * Retrieve table name
+     *
      * @param string $tableName
      * @return string
      */
@@ -431,6 +469,8 @@ class IndexBuilder
     }
 
     /**
+     * Update rule product data
+     *
      * @param Rule $rule
      * @return $this
      * @deprecated 101.0.0
@@ -456,6 +496,8 @@ class IndexBuilder
     }
 
     /**
+     * Apply all rules
+     *
      * @param Product|null $product
      * @throws \Exception
      * @return $this
@@ -495,8 +537,10 @@ class IndexBuilder
     }
 
     /**
+     * Calculate rule product price
+     *
      * @param array $ruleData
-     * @param null $productData
+     * @param array $productData
      * @return float
      * @deprecated 101.0.0
      * @see ProductPriceCalculator::calculate
@@ -507,6 +551,8 @@ class IndexBuilder
     }
 
     /**
+     * Get rule products statement
+     *
      * @param int $websiteId
      * @param Product|null $product
      * @return \Zend_Db_Statement_Interface
@@ -520,6 +566,8 @@ class IndexBuilder
     }
 
     /**
+     * Save rule product prices
+     *
      * @param array $arrData
      * @return $this
      * @throws \Exception
@@ -535,7 +583,7 @@ class IndexBuilder
     /**
      * Get active rules
      *
-     * @return array
+     * @return RuleCollection
      */
     protected function getActiveRules()
     {
@@ -545,7 +593,7 @@ class IndexBuilder
     /**
      * Get active rules
      *
-     * @return array
+     * @return RuleCollection
      */
     protected function getAllRules()
     {
@@ -553,6 +601,8 @@ class IndexBuilder
     }
 
     /**
+     * Get product
+     *
      * @param int $productId
      * @return Product
      */
@@ -565,6 +615,8 @@ class IndexBuilder
     }
 
     /**
+     * Log critical exception
+     *
      * @param \Exception $e
      * @return void
      */
diff --git a/vendor/magento/module-catalog-rule/Model/Indexer/RuleProductPricesPersistor.php b/vendor/magento/module-catalog-rule/Model/Indexer/RuleProductPricesPersistor.php
index 0b1264a2162..25bcfb8f20e 100644
--- a/vendor/magento/module-catalog-rule/Model/Indexer/RuleProductPricesPersistor.php
+++ b/vendor/magento/module-catalog-rule/Model/Indexer/RuleProductPricesPersistor.php
@@ -76,25 +76,19 @@ class RuleProductPricesPersistor
             );
         }
 
-        $productIds = [];
-
-        try {
-            foreach ($priceData as $key => $data) {
-                $productIds['product_id'] = $data['product_id'];
-                $priceData[$key]['rule_date'] = $this->dateFormat->formatDate($data['rule_date'], false);
-                $priceData[$key]['latest_start_date'] = $this->dateFormat->formatDate(
-                    $data['latest_start_date'],
-                    false
-                );
-                $priceData[$key]['earliest_end_date'] = $this->dateFormat->formatDate(
-                    $data['earliest_end_date'],
-                    false
-                );
-            }
-            $connection->insertOnDuplicate($indexTable, $priceData);
-        } catch (\Exception $e) {
-            throw $e;
+        foreach ($priceData as $key => $data) {
+            $priceData[$key]['rule_date'] = $this->dateFormat->formatDate($data['rule_date'], false);
+            $priceData[$key]['latest_start_date'] = $this->dateFormat->formatDate(
+                $data['latest_start_date'],
+                false
+            );
+            $priceData[$key]['earliest_end_date'] = $this->dateFormat->formatDate(
+                $data['earliest_end_date'],
+                false
+            );
         }
+        $connection->insertOnDuplicate($indexTable, $priceData);
+
         return true;
     }
 }
