diff --git a/vendor/magento/framework/Mview/TriggerCleaner.php b/vendor/magento/framework/Mview/TriggerCleaner.php
index 81ccf9a49..aba8d5fd4 100644
--- a/vendor/magento/framework/Mview/TriggerCleaner.php
+++ b/vendor/magento/framework/Mview/TriggerCleaner.php
@@ -87,8 +87,10 @@ class TriggerCleaner
         $remainingTriggers = array_diff_key($this->DbTriggers, $this->processedTriggers);
         foreach ($remainingTriggers as $trigger) {
             $view = $this->createViewByTableName($trigger['EVENT_OBJECT_TABLE']);
-            $view->unsubscribe();
-            $view->getState()->delete();
+            if ($view->getActionClass()) {
+                $view->unsubscribe();
+                $view->getState()->delete();
+            }
         }

         return true;
